[14.0][FIX] intrastat_product: don't force wrong domain in moves

joao-p-marques opened the pull request:
The intrastat transaction domain is already forced in the model and in the field in account.move.
It was being incorrectly forced in the move, where it needed to be associated to the same company as the record, but those types can have no company associated

@Tecnativa
TT29582

ping @pedrobaeza @victoralmau

alexis-via commented:
There is no domain on the field definition, cf https://github.com/OCA/intrastat-extrastat/blob/14.0/intrastat_product/models/account_move.py#L12
It's true that intrastat.transaction can have company_id = False. So the domain should be ['|', ('company_id', '=', False), ('company_id', '=', company_id)]. But I don't think it's a good idea to remove the domain.

pedrobaeza commented:
Put the suggested domain, but on the field, the same as in:

https://github.com/odoo/odoo/blob/44e05cd802593f6d18dcdec41fe821e30d7ae893/addons/account/models/account_move.py#L211

joao-p-marques committed:
[FIX] intrastat_product: don't force wrong domain in moves

The intrastat transaction domain is already forced in the model and in the field in account.move.
It was being incorrectly forced in the move, where it needed to be associated to the same company as the record, but those types can have no company associated

TT29582

joao-p-marques commented:
Thanks @pedrobaeza @alexis-via 

Changes made :+1:

alexis-via commented:
/ocabot merge minor

pedrobaeza commented:
/ocabot merge patch

rousseldenis commented on intrastat_product/models/account_move.py:
```diff
@@ -16,6 +16,7 @@ class AccountMove(models.Model):
         tracking=True,
         check_company=True,
         help="Intrastat nature of transaction",
+        domain="['|', ('company_id', '=', False), ('company_id', '=', company_id)]",
```
@joao-p-marques @pedrobaeza Shouldn't this be :

`domain="['|', ('company_id', '=', False), ('company_id', 'in', company_ids)]"`

pedrobaeza replied:
This is not a record rule, but a domain to restrict the selectable values, and they should be according the company of the move. There's no sense on selecting something of another of the available companies for a move of a specific company.
